test/integration/query-interface/changeColumn.test.js

```diff
@@ -361,6 +362,131 @@ describe(Support.getTestDialectTeaser('QueryInterface'), () => {
         expect(refs[0].referencedTableName).to.equal('Users');
         expect(refs[0].referencedColumnName).to.equal('id');
       });
+
+      it('should retain ON UPDATE and ON DELETE constraints after a column is changed', async function () {
+        await this.queryInterface.createTable('users', {
+          id: {
+            type: DataTypes.INTEGER,
+            primaryKey: true,
+            autoIncrement: true,
+          },
+          name: {
+            type: DataTypes.STRING,
+            allowNull: true,
+          },
+          level_id: {
+            type: DataTypes.INTEGER,
+            allowNull: false,
+            references: {
+              key: 'id',
+              model: 'level',
+            },
+            onDelete: 'CASCADE',
+            onUpdate: 'CASCADE',
+          },
+        });
+
+        await this.queryInterface.createTable('level', {
+          id: {
+            type: DataTypes.INTEGER,
+            primaryKey: true,
+            autoIncrement: true,
+          },
+          name: {
+            type: DataTypes.CHAR,
+            allowNull: false,
+          },
+        });
+
+        await this.queryInterface.changeColumn('users', 'name', {
```

Comment:
Can you also (or instead) apply a change to the `level_id` column? (such as update its nullability and/or its name) to make sure it does not lose its reference?

Reply:
Added a change for level_id -> changed the nullability.